source/FreeRTOS_DNS_Parser.c

```diff
@@ -373,124 +420,189 @@
-                                    {
-                                        BaseType_t xOffset1, xOffset2;
+                        /* Make a copy of the end-point because xApplicationDNSQueryHook() is allowed
+                         * to write into it. */
+                        ( void ) memcpy( &( xEndPoint ), pxEndPoint, sizeof( xEndPoint ) );
 
-                                        xOffset1 = ( BaseType_t ) ( pucByte - pucUDPPayloadBuffer );
-                                        xOffset2 = ( BaseType_t ) ( ( ( uint8_t * ) pcRequestedName ) - pucUDPPayloadBuffer );
+                        #if ( ipconfigUSE_IPv6 != 0 )
+                            {
+                                /*logging*/
+                                FreeRTOS_printf( ( "prvParseDNS_HandleLLMNRRequest[%s]: type %04X\n", pxSet->pcName, pxSet->usType ) );
 
-                                        pxNetworkBuffer = pxNewBuffer;
-                                        pucNewBuffer = &( pxNetworkBuffer->pucEthernetBuffer[ ipUDP_PAYLOAD_OFFSET_IPv4 ] );
+                                xEndPoint.usDNSType = xSet.usType;
+                            }
+                        #endif /* ( ipconfigUSE_IPv6 != 0 ) */
 
-                                        pucByte = &( pucNewBuffer[ xOffset1 ] );
-                                        pcRequestedName = ( char * ) &( pucNewBuffer[ xOffset2 ] );
-                                        pxDNSMessageHeader = ( ( DNSMessage_t * ) pucNewBuffer );
+                        /* If this is not a reply to our DNS request, it might an LLMNR
+                         * request. */
+                        if( xApplicationDNSQueryHook( &xEndPoint, xSet.pcName ) )
+                        {
+                            != pdFALSE )
+                            {
+                                int16_t usLength;
+                                NetworkBufferDescriptor_t * pxNewBuffer = NULL;
+                                LLMNRAnswer_t * pxAnswer;
+                                uint8_t * pucNewBuffer = NULL;
+
+                                if( pxNetworkBuffer != NULL )
+                                {
+                                    if( xBufferAllocFixedSize == pdFALSE )
+                                    {
+                                        size_t uxDataLength = uxBufferLength +
+                                                              sizeof( UDPHeader_t ) +
+                                                              sizeof( EthernetHeader_t ) +
+                                                              uxIPHeaderSizePacket( pxNetworkBuffer );
+
+                                        #if ( ipconfigUSE_IPv6 != 0 )
+                                            if( xSet.usType == dnsTYPE_AAAA_HOST )
+                                            {
+                                                uxExtraLength = sizeof( LLMNRAnswer_t ) + ipSIZE_OF_IPv6_ADDRESS - sizeof( pxAnswer->ulIPAddress );
```

Comment:
The declaration of `uxExtraLength` was dropped?

Reply:
Strange, how is it not giving compilation error. Will update.